packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -395,9 +410,17 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         // without breaking the current interface.
         bytes memory opaqueData = abi.encodePacked(msg.value, _value, _gasLimit, _isCreation, _data);
 
+        // Compute the updated global hash union of deposit data and cache the previous value on the stack for
+        // event emission.
+        bytes32 unionBefore = depositHashUnion;
+        bytes32 unionAfter = keccak256(abi.encode(unionBefore, from, _to, DEPOSIT_VERSION, opaqueData));
+
         // Emit a TransactionDeposited event so that the rollup node can derive a deposit
         // transaction for this deposit.
-        emit TransactionDeposited(from, _to, DEPOSIT_VERSION, opaqueData);
+        emit TransactionDeposited(from, _to, DEPOSIT_VERSION, opaqueData, unionBefore, unionAfter);
```

Comment:
Derivation decoding is very strict, we may need to update that in a prior hardfork. Or capture the `unionBefore` and `unionAfter` attributes in a separate event. 
See https://github.com/ethereum-optimism/optimism/blob/72166a682d00112ff97b5fc104b897401e5d9950/op-node/rollup/derive/deposit_log.go#L69

Reply:
Rather than dealing with backwards compatibility here, I think it's more pragmatic to add a new event. Added `HashUnionUpdated`.